base/complex.jl

```diff
@@ -1028,3 +1028,68 @@ function complex(A::AbstractArray{T}) where T
     end
     convert(AbstractArray{typeof(complex(zero(T)))}, A)
 end
+
+function div(a::Complex{T}, b::Complex{V}, r::RoundingMode=RoundNearest) where {T<:Integer, V<:Integer}
+    R = promote_type(T, V)
+    a, b = Complex{R}(a), Complex{R}(b)
```

Comment:
I think, it would be better to use dispatch for promotion here, since that way, it's more transparent, if a user wants to have a different implementation for a custom integer type.